### ISSUE TITLE
Why radio button is always required

### ISSUE THREAD

**Issue by kkaminski-github** [json-editor/json-editor#1013]
Hi!

I was wondering why radio editor is forcing required option in preBuild method:
```javascript
this.schema.required = true /* force editor into required mode to prevent creation of empty radio button */
```

I have a problem, when in my schema there is a radio button, I am forced to fill it with value. In my case possibility to not having this key-value pair in resulting object is also important. I have tried by commenting this line out and it works just fine. Therefore my question - what is this mysterious empty radio button? 

```javascript
{
  "title": "json schema example",
  "type": "object",
  "properties": {
    "radio": {
          "type": "string",
          "format": "radio",
          "title": "Radio Buttons", 
          "enum": ["Home", "Work"]
        },
  }
}
```

Best regards,
Krzysztof

**Comment by germanbisurgi** [comment]
Good point. I don't think that should be the case. When i commented that line this happened:

<img width="813" alt="Bildschirmfoto 2021-06-25 um 09 16 16" src="https://user-images.githubusercontent.com/13135260/123386000-78987480-d596-11eb-98cd-fafea5975630.png">

And i also could use the `show_opt_it` option too, that would not be possible  otherwise:

<img width="767" alt="Bildschirmfoto 2021-06-25 um 09 22 30" src="https://user-images.githubusercontent.com/13135260/123386380-e5ac0a00-d596-11eb-9cdb-41d8ad7e0e0c.png">

I am working on a fix

**Comment by kkaminski-github** [comment]
Thank you for your answer,

I need to point that in my case commenting out that line didn't cause this empty radio button to appear. 
I have slightly changed SelectEditor:
![image](https://user-images.githubusercontent.com/55243725/123396582-c1095f80-d5a1-11eb-8595-a8657150f174.png)

In my case you need to opt in for undefined option in select editor.

Might be helpful ;)

Best regards,
Krzysztof

**Comment by germanbisurgi** [comment]
That could be a good feature request :)

**Comment by schmunk42** [comment]
fixed in #1014
